libraries/lwIP_Ethernet/examples/EthClient/EthClient.ino

```diff
@@ -1,57 +1,50 @@
 /*
     This sketch establishes a TCP connection to a "quote of the day" service.
     It sends a "hello" message, and then prints received data.
+
+    This is Ethernet version of:
+    https://github.com/esp8266/Arduino/blob/master/libraries/ESP8266WiFi/examples/WiFiClient/WiFiClient.ino
 */
 
-#include <SPI.h>
-#include <W5500lwIP.h>
-//or #include <W5100lwIP.h>
-//or #include <ENC28J60lwIP.h>
+#include <LwipEthernet.h>
 
-#include <WiFiClient.h> // WiFiClient (-> TCPClient)
+Wiznet5500lwIP eth(/*SS*/16);     // <== adapt to your hardware
 
 const char* host = "djxmmx.net";
 const uint16_t port = 17;
 
-using TCPClient = WiFiClient;
-
-#define CSPIN 16 // wemos/lolin/nodemcu D0
-Wiznet5500lwIP eth(CSPIN);
-
 void setup() {
   Serial.begin(115200);
 
-  SPI.begin();
-  SPI.setClockDivider(SPI_CLOCK_DIV4); // 4 MHz?
-  SPI.setBitOrder(MSBFIRST);
-  SPI.setDataMode(SPI_MODE0);
-  eth.setDefault(); // use ethernet for default route
-  if (!eth.begin()) {
-    Serial.println("ethernet hardware not found ... sleeping");
+  Serial.println("\nEthernet\n");
+
+  eth.setDefault(true); // default route set through this interface
```

Comment:
could we avoid setDefault() for ethernet classes, perhaps via 2nd template param of the LwipIntfDev?
*(this may have been already discussed, though, but I really don't remember the outcome of that)*

Reply:
These examples now have `eth.setDefault()` commented with an explanation.